lms/djangoapps/mobile_api/users/views.py

```diff
@@ -666,6 +676,39 @@ def _get_course_ids_where_user_has_completions(
 
         return set(context_keys)
 
+    def _get_course_image_url(self, course_overview):
+        """
+        Returns full path to course image.
+        """
+        if not course_overview.course_image_url:
+            return None
+
+        return f"{settings.LMS_ROOT_URL}{course_overview.course_image_url}"
+
+    def _get_course_total_sizes(self, enrollments):
+        """
+        Retrieve course sizes from cache or database and return a complete mapping.
+        """
+        course_ids = [str(enrollment.course_overview.id) for enrollment in enrollments]
+
+        cache_keys = {course_id: COURSE_SIZE_CACHE_KEY.format(course_id=course_id) for course_id in course_ids}
+        cached_sizes = cache.get_many(cache_keys.values())
+
+        course_sizes = {course_id: cached_sizes[cache_keys[course_id]] for course_id in course_ids if cache_keys[course_id] in cached_sizes}
+        not_cached_course_ids = [course_id for course_id in course_ids if course_id not in course_sizes]
+
+        if not_cached_course_ids:
+            db_sizes = OfflineBlockSize.objects.filter(course_id__in=not_cached_course_ids) \
```

Comment:
```suggestion
            db_sizes = OfflineBlockSize.objects.filter(course_id__in=not_cached_course_ids).values(
                'course_id'
            ).annotate(total_size=Sum('size'))
```
I think it will be more readable.

Reply:
Also, I think it's better to name the variable db_course_sizes.